int32 center_x
int32 center_y
int32 rad
uint8[] im
